src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1679,6 +1679,61 @@ class TransformClauseSegment(BaseSegment):
     )
 
 
+@spark3_dialect.segment(replace=True)
+class ExplainStatementSegment(BaseSegment):
+    """An `Explain` statement.
+
+    Enhanced from ANSI dialect to allow for additonal parameters.
+
+    EXPLAIN [ EXTENDED | CODEGEN | COST | FORMATTED ] explainable_stmt
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-explain.html
+    """
+
+    type = "explain_statement"
+
+    explainable_stmt = ansi_dialect.get_segment(
+        "ExplainStatementSegment"
+    ).explainable_stmt.copy(
+        insert=[
+            Ref("AlterDatabaseStatementSegment"),
+            Ref("AlterTableStatementSegment"),
+            Ref("AlterViewStatementSegment"),
+            Ref("CreateDatabaseStatementSegment"),
+            Ref("CreateFunctionStatementSegment"),
+            Ref("CreateTableStatementSegment"),
+            Ref("CreateViewStatementSegment"),
+            ansi_dialect.get_segment("DropDatabaseStatementSegment"),
+            Ref("DropFunctionStatementSegment"),
+            ansi_dialect.get_segment(
+                "DropViewStatementSegment",
+            ),
+            Ref("UseDatabaseStatementSegment"),
+            Ref("TruncateStatementSegment"),
+            Ref("MsckRepairTableStatementSegment"),
+            Ref("RefreshTableStatementSegment"),
+            Ref("RefreshFunctionStatementSegment"),
+            Ref("LoadDataSegment"),
+            Ref("InsertOverwriteDirectorySegment"),
+            Ref("InsertOverwriteDirectoryHiveFmtSegment"),
+        ],
+    )
+
+    match_grammar = StartsWith("EXPLAIN")
+
+    parse_grammar = Sequence(
+        "EXPLAIN",
```

Comment:
This does nothing as they both start with `EXPLAIN`. See it's like this in ANSI but still think we should clean up (would take the ANSI clean up at same time if you like!)

```suggestion
    match_grammar = Sequence(
        "EXPLAIN",
```

Reply:
I'll clean this up in Spark and ANSI